packages/ra-core/src/controller/EditController.tsx

```diff
@@ -133,9 +130,13 @@ const EditController = (props: Props) => {
                 dispatch(updateAction);
             }
         },
-        [id, resource, basePath, record]
+        [resource, id, record, basePath, undoable, dispatch]
     );
 
+    if (!children) {
```

Comment:
Not really sure about this one but the lint rule for hook complained about us applying hooks conditionnaly

Reply:
I'd say ignore the lint rule on that one

Reply:
Why are we doing this though? Shouldn't we throw an error about it ?

Reply:
Ok but we are breaking the hooks rules here

Reply:
Btw, couldn't we fallback to the guesser when we detect there are no children ?

Reply:
I agree, this is not a normal use case, we should throw. And mark the `children` prop as required.

Also, we can't fall back to an `EditGuesser` because there is no such thing in `ra-core` (it's in `ra-ui-material-ui`)